### PR TITLE
Architecture poc

### PR DESCRIPTION
**Pull request by odrusso**

### CODE REVIEW

File: app/questions.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import sys
+from sympy import symbols, latex, Eq, Rational
+
+QUESTIONS = {
+    'mcat': set(
+        '1',
+    ),
+}
+
+def get_question(question_id, question_type, seed):
```
Comment (odrusso) [inline app/questions.py:10]: I think this might be a good use case for an `enum`

File: app/questions.py [hunk 2]
```diff
@@ -1,28 +1,56 @@
-import sys
-from sympy import symbols, latex, Eq, Rational
+from sympy import symbols, latex, Eq
 
-QUESTIONS = {
-    'mcat': set(
-        '1',
-    ),
-}
+class QuestionError:
```
Comment (odrusso) [inline app/questions.py:3]: This could inherit from the Exception/Error class I think

File: app/questions.py [hunk 3]
```diff
@@ -1,28 +1,56 @@
...
+
+
+class Question:
+    def __init__(self, seed):
+        self.description = ""
+        self.question = ""
+        self.seed = seed
+    
+    @property
+    def __dict__(self):
```
Comment (odrusso) [inline app/questions.py:20]: I vote this should be called something nicer

File: app/questions.py [hunk 4]
```diff
@@ -1,28 +1,56 @@
...
+
+        x = symbols("x")
+        ## TODO: Use seed
+        a = 2
+        b = 3
+        self.description = "Solve a linear equation."
+        self.question = latex(Eq(a * x + b, 0))
+
+
+QUESTION_MAPPING = {
```
Comment (odrusso) [inline app/questions.py:39]: I like it

File: app/questions.py (hunk 2 above)
Comment (ashtonmoomoo) [reply app/questions.py:3]: Agree

File: app/questions.py (hunk 3 above)
Comment (ashtonmoomoo) [reply app/questions.py:20]: something like `json(self)`?

File: app/app.py [hunk 5]
```diff
@@ -10,7 +10,7 @@ def question_router(question_type, question_id):
     seed = request.args.get("seed", time())
     question = get_question(question_type, question_id, seed)
 
-    return question.__dict__
+    return question.json
```
Comment (odrusso) [inline app/app.py:13]: Should this be `question.json()`?

Comment (odrusso) [reply app/app.py:13]: i.e. actually call the function

Comment (ashtonmoomoo) [reply app/app.py:13]: nah it's a class property, not a function

File: app/questions.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        '1': MCATQuestion1,
+    },
+}
+
+
+def get_question(question_type, question_id, seed):
+
+    if QUESTION_MAPPING.get(question_type, None):
+        if QUESTION_MAPPING.get(question_type).get(question_id, None):
+            question = QUESTION_MAPPING.get(question_type).get(question_id)(seed)
```
Comment (odrusso) [inline app/questions.py:62]:
```suggestion
    if questionType := QUESTION_MAPPING.get(question_type, None):
        if question := questionType.get(question_id, None):
```
This is ur style isn't it

Comment (ashtonmoomoo) [reply app/questions.py:62]: brain obvs went blank when I was writing this lol, you're absolutely right. Except we use snake_case round these parts ;)

**Comment by odrusso** [comment]
Can't review because I made the PR but LGTM

**Review by ashtonmoomoo — APPROVED** [review]
yeah nice

**Pull request merged**